File: MattEland.Emergence.Domain/Firewall.fs
namespace MattEland.Emergence.Domain

type Firewall (position: Position) = 
  inherit WorldObject(position, System.Guid.NewGuid())
  
  let mutable isOpen: bool = false
  
  member this.IsOpen = isOpen

  override this.AsciiCharacter =
    match isOpen with 
    | true -> ':'
    | _ -> '|'

  override this.ZIndex = 75

  interface IInteractive with
    member this.interact source =
      seq {
        yield new DisplayMessage("The firewall is not open.")
      }

File: MattEland.Emergence.Model/Entities/Firewall.cs
using System;
using System.Collections.Generic;
using MattEland.Emergence.Model.Messages;

namespace MattEland.Emergence.Model.Entities
{
    public class Firewall : WorldObject, IInteractive
    {
        public Firewall(Position pos) : base(pos, Guid.NewGuid())
        {
        }

        public override string ForegroundColor => IsOpen ? GameColors.LightGreen : GameColors.Orange;

        public bool IsOpen { get; set; }

        public override char AsciiChar => IsOpen ? ':' : '|';

        public override int ZIndex => 50;
        public void Interact(ICommandContext context)
        {
            if (IsOpen)
            {
                context.MoveExecutingActor(Pos);
            }
            else
            {
                context.DisplayText("The firewall is still up.");
            }
        }
    }
}